sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -164,4 +166,18 @@ object SQLExecution {
       }
     }
   }
+
+  /**
+   * Wrap passed function to ensure sparkContext local properties are forwarded to execution thread
+   */
+  def withThreadLocalCaptured[T](sparkSession: SparkSession)(body: => T)(
+    exec: ExecutionContext): Future[T] = {
+    val activeSession = sparkSession
+    val localProps = Utils.cloneProperties(sparkSession.sparkContext.getLocalProperties)
```

Comment:
do we have to clone? I know the local properties are mutable, but that was the behavior until now.

Reply:
This is just to ensure case when the future is in waiting due to `ExecutorService` threads being fully occupied and meanwhile any changes happen in localProperties, that do not affect the properties which were passed on submission. Do you feel its reasonable.?

Reply:
Yea I think it's reasonable, but a little worried about performance. @hvanhovell what do you think?

Reply:
I would be in favor of cloning them (BTW I am not sure why Utils.cloneProperties does not just use clone - the current implementation is probably less efficient). My main worry is that the local properties object is shared between multiple threads (which can happen if have multiple broadcast joins for example), and that these threads no start influencing each other and the main thread; I would like to avoid midnight debugging sessions caused by this.